src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs

```diff
@@ -59,9 +59,9 @@ public AcquireTokenForManagedIdentityParameterBuilder AcquireTokenForManagedIden
         /// Detects and returns the managed identity source available on the environment.
         /// </summary>
         /// <returns>Managed identity source detected on the environment if any.</returns>
-        public static ManagedIdentitySource GetManagedIdentitySource()
+        public static async Task<ManagedIdentitySource> GetManagedIdentitySourceAsync()
```

Comment:
This is a breaking change. We cannot rename the public API.

Reply:
made a new method and made this one obsolete